GHC 8.2.1 support.

robinp opened the pull request:
@ivan444 please review.
@mpickering FYI.

FYI I didn't purge version macros from `Ghc.hs` entirely, only where the signal-to-noise ratio seemed justified. Also, I didn't touch record-related logic, since that might change depending on #19.

ivan444 commented on haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs:
```diff
@@ -243,15 +230,17 @@ declsFromRenamed ctx (hsGroup, _, _, _) =
         namesFromHsIbWc :: LHsSigWcType Name -> [Name]
-        namesFromHsIbWc (HsIB names _) = names
+        namesFromHsIbWc =
+#if __GLASGOW_HASKELL__ >= 802
```
Could this also go to Compat.hs?

robinp commented:
I added a few more to Compat, but a few in-flux parts still remain (see updated PR desc above). CI had a bad time, should be fine now.

Also added `run-ghc-tests.sh` locally which runs for all GHCs both the unit tests and Kythe verifier tests.
